### PR TITLE
Test/location space hall endpoints

### LINKED ISSUE #114: Tests for API Endpoints
## API Endpoints

Track implementation and integration test coverage for all endpoints.
- [x] `Locations`
- [x] `Halls`
- [x] `Spaces`
- [x] `Productions`
- [ ] `Events`
- [ ] `Artists`
- [ ] `Blogposts`
- [ ] `Collections`
- [x] `Auth`
- [x] `Admin`

### PR DESCRIPTION
**Pull request by jarnex47**
- Fix repositories for locations, spaces, and halls to return **404** when deleting a non-existent resource
- Add **Serialize** on **PostPayload** types for location, space, and hall DTOs (needed for test helpers)
- Add CRUD integration tests for the **locations, spaces**, and **halls** endpoints, covering success and not-found cases for GET, POST, PUT, and DELETE

### CODE REVIEW

File: backend/tests/fixtures/halls.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    'zaal-b',
+    '20000000-0000-0000-0000-000000000002'
+),
+(
+    '30000000-0000-0000-0000-000000000003',
+    3003,
+    'Studio Hal',
+    'studio-hal',
+    '20000000-0000-0000-0000-000000000003'
+);
```
Comment (hanneskt) [inline backend/tests/fixtures/halls.sql:29]: suggestion: assign at least 2 halls to one of the spaces, and have a space without any halls

In the fixtures for m-1 relations, I try to make sure there is one of each case: 0-1, 1-1 and >=2-1. This often catches weird edge cases.

I would add this now, but this will probably become more relevant when we return joined objects (a space returns a list of it's halls as well, for example) where we can use this to test the different cases in the object above (a space with 2, 1 and 0 halls). We can test for it whenever we add that feature

File: backend/tests/fixtures/spaces.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    2002,
+    'Kleine Zaal',
+    '10000000-0000-0000-0000-000000000001'
+),
+(
+    '20000000-0000-0000-0000-000000000003',
+    2003,
+    'Studio',
+    '10000000-0000-0000-0000-000000000002'
+);
```
Comment (hanneskt) [inline backend/tests/fixtures/spaces.sql:19]: praise: you did here what I talked about, each case in the m-1 relation is covered!

**Review by hanneskt — CHANGES_REQUESTED** [review]
Basically good. Just have the fixture cover the different cases.

The tests are well written otherwise!

File: backend/tests/fixtures/halls.sql (hunk 1 above)
Comment (jarnex47) [reply backend/tests/fixtures/halls.sql:29]: alright, will do!

**Review by hanneskt — APPROVED** [review]
Alright great!

**Review by Kaya-Sem — APPROVED** [review]
